src/FilterPluginManagerFactory.php

```diff
@@ -6,26 +6,22 @@
 
 use Interop\Container\ContainerInterface;
 use Laminas\ServiceManager\Config;
-use Laminas\ServiceManager\FactoryInterface;
-use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
 
 use function is_array;
 
 class FilterPluginManagerFactory implements FactoryInterface
 {
-    /**
-     * laminas-servicemanager v2 support for invocation options.
-     *
-     * @param array
-     */
-    protected $creationOptions;
```

Comment:
Hmm, with this being a non-`final` class, this change is probably to be considered a BC break.

Same applies to signature changes

Reply:
Scheduling time to review tomorrow (2022-10-13)

Reply:
thank you both for your time.

Happy to add back the v2 API stuff that i removed due to the comments above, if that avoids a BC break.